extensions/apiconnect/src/types/oas-extensions/x-ibm-configuration/assembly/execute/invoke.policy.ts

```diff
@@ -0,0 +1,126 @@
+export type InvokePolicy = DataPowerGateway | DataPowerAPIGateway;
+
+export type DataPowerGateway = V100;
+export type DataPowerAPIGateway = V200 | V210 | V220 | V230;
+export interface V100 {
+  version: '1.0.0';
+  title?: string;
+  description?: string;
+  'target-url': string;
+  'tls-profile'?: string;
+  /**
+   * @defaultValue `'GET'`
+   */
+  verb?: 'GET' | 'POST' | 'PUT' | 'DELETE' | 'PATCH' | 'HEAD' | 'OPTIONS';
+  /**
+   * @defaultValue `60`
+   */
+  timeout?: number;
+  /**
+   * @defaultValue `false`
+   */
+  compression?: boolean;
+  username?: string;
+  password?: string;
+  output?: string;
+  'cache-key'?: string;
+  /**
+   * @defaultValue `'protocol'`
+   */
+  'cache-response'?: 'protocol' | 'no-cache' | 'time-to-live';
+  /**
+   * @defaultValue `false`
+   */
+  'cache-putpost-response'?: boolean;
+  /**
+   * @remarks
+   * Valid range: 5-31708800
+   * @defaultValue `900`
+   */
+  'cache-ttl'?: number;
+  'stop-on-error'?: string[];
+}
+
+export interface V200 extends Omit<V100, 'version'> {
+  version: '2.0.0';
+  /**
+   * @defaultValue `'detect'`
+   */
+  'backend-type'?: 'detect' | 'xml' | 'json' | 'binary' | 'graphql';
+  /**
+   * @defaultValue `false`
+   */
+  'inject-proxy-headers'?: boolean;
+  /**
+   * @defaultValue `false`
+   */
+  'decode-request-params'?: boolean;
+  /**
+   * @defaultValue `false`
+   */
+  'encode-plus-char'?: boolean;
+  /**
+   * @defaultValue `false`
+   */
+  'keep-payload'?: boolean;
+  /**
+   * @defaultValue `false`
+   */
+  'use-http-10'?: boolean;
+  /**
+   * @defaultValue `true`
+   */
+  'chunked-uploads'?: boolean;
+  /**
+   * @defaultValue
+   * ```typescript
+   * {
+   *   type: 'blacklist',
+   *   values: [],
+   * }
+   * ```
+   */
+  'header-control'?: {
+    type: 'blacklist' | 'whitelist';
+    values: string[];
+  };
+  /**
+   * @defaultValue
+   * ```typescript
+   * {
+   *   type: 'whitelist',
+   *   values: [],
+   * }
+   * ```
+   */
+  'parameter-control'?: {
+    type: 'blacklist' | 'whitelist';
+    values: string[];
+  };
+  'follow-redirect'?: boolean;
+}
+
+export interface V210 extends Omit<V200, 'version'> {
+  version: '2.1.0';
+  'http-version': 'HTTP/1.0' | 'HTTP/1.1' | 'HTTP/2';
+  'http2-required'?: boolean;
+  'websocket-upgrade'?: boolean;
```

Comment:
Need to deprecate. Apparently this is replaced by `webscket-assembly`

From https://www.ibm.com/docs/en/api-connect/10.0.x?topic=execute-invoke:
> Note:
> 
>  This property is available only for GraphQL APIs.
>  The websocket-upgrade property is deprecated. To allow WebSocket upgrade requests that can manage API processing data, define an assembly WebSocket upgrade policy. For more information, see websocket-upgrade.

From https://www.ibm.com/docs/en/api-connect/10.0.x?topic=10030-whats-new-in-version-10020:

> New Websocket Upgrade assembly policy
> Use the GraphQL Websocket Upgrade policy to process API requests and responses through a WebSocket connection. This policy is supported only for a GraphQL API. For more information, see GraphQL Websocket Upgrade.
> Note: The Allow WebSocket Upgrade setting on the Invoke policy is deprecated in favor of this new policy.
>

Reply:
Also, this can only be defined under certain conditions.

#TODO: Update this typedef to reflect this condition.

From: https://www.ibm.com/docs/en/api-connect/10.0.x?topic=execute-invoke:

> Yes, if backend-type is set to graphql or detect, and verb is set to POST or keep.

This is already defined in V220 typedef, need to move it here.

Reply:
The docs doesn't clearly state where WebSocket Upgrade should be located at, but it seems like it should either be under `execute` or `assembly`?

This docs only mentions `execute`:

https://www.ibm.com/docs/en/api-connect/10.0.x?topic=file-extensions-openapi-specification

Reply:
Upon re-reading my first comment in this comment chain, this apparently belongs under `assembly`.

Reply:
Yet another update: It seems like the API Connect documentation refers to policies in `execute` as "assembly policies". Hence I've shifted `websocket-upgrade` back into `execute`.

For example, in https://www.ibm.com/docs/en/api-connect/2018.x?topic=cli-openapi-30-support-in-api-connect:

> ...
> - Assembly policies.
> 
>     - Only the following policies are supported:
>  ...